src/csharp/Microsoft.Spark/Utils/PythonSerDe.cs

```diff
@@ -67,5 +68,57 @@ internal static object[] GetUnpickledObjects(Stream stream, int messageLength)
                 ArrayPool<byte>.Shared.Return(buffer);
             }
         }
+
+        /// <summary>
+        /// Custom pickler for GenericRow objects.
+        /// Refer to
+        /// spark/sql/core/src/main/scala/org/apache/spark/sql/execution/python/EvaluatePython.scala
+        /// </summary>
+        internal class GenericRowPickler : IObjectPickler
+        {
+            private readonly string _module = "pyspark.sql.types";
+
+            public void Register()
+            {
+                Pickler.registerCustomPickler(GetType(), this);
+                Pickler.registerCustomPickler(typeof(GenericRow), this);
+            }
+
+            public void pickle(object o, Stream stream, Pickler currentPickler)
+            {
+                if (o.Equals(this))
+                {
+                    SerDe.Write(stream, Opcodes.GLOBAL);
+                    SerDe.Write(stream, Encoding.UTF8.GetBytes(
+                        $"{_module}\n_create_row_inbound_converter\n"));
+                }
+                else
+                {
+                    if (!(o is GenericRow genericRow))
+                    {
+                        throw new InvalidOperationException("A GenericRow object is expected.");
```

Comment:
In this exception can we add message for what is actually being passed instead of GenericRow?

Reply:
Removed. Thanks!